ports/zlib/vcpkg.json

```diff
@@ -12,6 +12,7 @@
     {
       "name": "vcpkg-cmake-config",
       "host": true
-    }
+    },
+    "zlib-ng"
```

Comment:
Don't forget to remove vcpkg-cmake and vcpkg-cmake-config.

Reply:
The show-stopper might be the lack of activating `ZLIB_COMPAT`. AFAICS it is not (and cannot be) mapped to a feature, and so it would need to be injected by triplet file customization.

Reply:
That's disappointing. The whole point of zlib-ng is to have a drop in replacement.

You have an example on how one would inject it?

Reply:
Looks like https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_cmake_configure_options

Reply:
"Drop-in replacement" is the same as "alternative", and vcpkg is unable to allow alternatives.

Basically, all (overlay) triplet files need
~~~
if(PORT STREQUAL "zlib-ng")
    set(VCPKG_CMAKE_CONFIGURE_OPTIONS "-DZLIB_COMPAT=ON")
endif()
~~~

(There are other efforts, but I don't think anything is ready now.)